swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -28,76 +26,134 @@ This guide will walk you through creating a RESTful API with two endpoints. In t
 
-```
-.
-├── covid19
-│   ├── Ballerina.toml
-│   └── service.bal
-```
+    Created new package 'restful_service' at /Users/restful-service.
+    ```
 
-- `Ballerina.toml` is the file that makes the folder a Ballerina package. It also contains a test directory to include tests for the service. However, this will not be used in this guide. 
-- The `service.bal` template file provides a look and feel about Ballerina services. 
+    This creates a directory named `restful-service` with sample Ballerina code, as shown below. 
+
+    ```
+    restful-service
+    ├── Ballerina.toml
+    └── main.bal
+    ```
+
+    - `Ballerina.toml` is the file that makes the folder a Ballerina package. 
+    - The `main.bal` file provides the look and feel of a simple Ballerina program. 
+
+2. Open the created package in VS Code.
+
+3. Remove the generated content in the `main.bal` file and open the diagram view in VS Code.
+
+    ![Open diagram view](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/open-diagram-view.gif)
 
 ## Create the dataset
 
-To keep things simple, an in-memory dataset is used with three entries. To add the definition of the record and the declaration of the table, replace the API template file (i.e., `service.bal`) with the code below.
+An in-memory dataset with three entries is used to keep things simple. Follow the steps below to add the definition of the record and the declaration of the [table](/learn/by-example/table/) that holds the data.
+
+1. Generate the record type corresponding to the request payload of the REST service by providing `CovidEntry` as the name and the sample JSON payload below.
+
+    >**Note:** You need to update the generated record by adding the pipe symbol to mark the record as a [closed one](https://ballerina.io/learn/by-example/controlling-openness/) and adding the `readonly` descriptor to the `iso_code` field to make it non-modifiable in order to use it as the key of the table.
+
+    ```json
+    {
+        "iso_code": "AFG",
+        "country": "Afghanistan",
+        "cases": 159303,
+        "deaths": 7386,
+        "recovered": 146084,
+        "active": 5833
+    }
+    ```
+
+    ![Create data record](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-record.gif)
+
+    The generated record type will be as follows.
+
+    ```ballerina
+    type CovidEntry record {|
+        readonly string iso_code;
+        string country;
+        int cases;
+        int deaths;
+        int recovered;
+        int active;
+    |};
+    ```
+
+2. Create the table, as shown below.
+
+    >**Note:** Select the `final` modifier and enter `CovidEntry` as the row type corresponding to the type of the members of the table, `iso_code` as the key, and `covidTable` as the variable name when creating the table.
+
+    ![Create data table](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-table.gif)
```

Comment:
There seems to be an issue with this GIF at the end. Shows erroneous code on the table member type.

Reply:
Updated.